### DISCUSSION TITLE
Does the Virtual Tour Plugin use Magnetic North or True North?

### DISCUSSION THREAD

**Discussion by AndyBaird** [mistic100/Photo-Sphere-Viewer#1472] [General]
I loaded in two sets of images and noticed one sets markers all seemed to be place 20 degrees off from where I expected them. In that set the metadata specified `GPS Img Direction Ref` was Magnetic North. So my theory is to just detect that and offset everything by 20 degrees but I was hoping someone might be able to confirm if the plugin use True North for the placement of markers?

**Comment by mistic100** [comment]
No idea. I use the data as it without altering them.

Most of the math is here https://github.com/mistic100/Photo-Sphere-Viewer/blob/main/packages/virtual-tour-plugin/src/utils.ts

My reference is http://www.movable-type.co.uk/scripts/latlong.html

**Reply by AndyBaird** [reply]
On the movable type docs I found this in the footnotes:

> All bearings are with respect to true north, 0°=N, 90°=E, etc; if you are working from a compass, magnetic north varies from true north in a complex way around the earth, and the difference has to be compensated for by variances indicated on local maps.

So for anyone else's reference, if your camera data uses magnetic north, you have to account for that and there are libraries suchs as [geomagJS](https://github.com/cmweiss/geomagJS) that can help.
